src/Queue/QueueServiceProvider.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Queue;
+
+use Flarum\Console\Event\Configuring;
+use Flarum\Foundation\AbstractServiceProvider;
+use Illuminate\Contracts\Debug\ExceptionHandler;
+use Illuminate\Contracts\Queue\Factory;
+use Illuminate\Queue\Connectors\ConnectorInterface;
+use Illuminate\Queue\Connectors\SyncConnector;
+use Illuminate\Queue\Console as Commands;
+use Illuminate\Queue\QueueManager;
+use Illuminate\Queue\Worker;
+
+class QueueServiceProvider extends AbstractServiceProvider
+{
+    protected $commands = [
+        Commands\FailedTableCommand::class,
+        Commands\FlushFailedCommand::class,
+        Commands\ForgetFailedCommand::class,
+        Commands\ListenCommand::class,
+        Commands\ListFailedCommand::class,
+        Commands\RestartCommand::class,
+        Commands\RetryCommand::class,
+        Commands\TableCommand::class,
+        Commands\WorkCommand::class,
+    ];
+
+    public function register()
+    {
+        $this->app->singleton(ConnectorInterface::class, function ($app) {
+            return $app['queue']->connection();
+        });
+
+        $this->app->singleton(Factory::class, function ($app) {
+            $manager = new QueueManager($app);
```

Comment:
Do we really benefit from the manager here? I don't think we need to support multiple queue backends, and using the manager just means that we'll have to translate extension configuration (e.g. settings in a GUI) into Laravel config and let the manager turn that into a connector instance. Why not turn our settings into a driver directly?

Reply:
Makes sense, didn't even consider that. We don't even need the configuration, any listener that binds into the resolving of this binding can change what queue driver is being used.

Reply:
Okay so simplifying the binding doesn't work, because the Mailer instance expects a Factory, not an implementation..

Reply:
The Validator too. I've reverted my latest commits. Please review again @flarum/core .